frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -126,31 +126,62 @@ function FileManagerMenu:setUpdateItemTable()
         checked_func = function() return self.ui.file_chooser.show_unsupported end,
         callback = function() self.ui:toggleUnsupportedFiles() end
     }
-    self.menu_items.items_per_page = {
-        text = _("Items per page"),
-        help_text = _([[This sets the number of items per page in:
+    self.menu_items.items = {
+        text = _("Items"),
+        sub_item_table = {
+            {
+                text = _("Items per page"),
+                help_text = _([[This sets the number of items per page in:
 - File browser and history in 'classic' display mode
 - File and directory selection
 - Table of contents
 - Bookmarks list]]),
-        keep_menu_open = true,
-        callback = function()
-            local SpinWidget = require("ui/widget/spinwidget")
-            local curr_items = G_reader_settings:readSetting("items_per_page") or 14
-            local items = SpinWidget:new{
-                width = Screen:getWidth() * 0.6,
-                value = curr_items,
-                value_min = 6,
-                value_max = 24,
-                ok_text = _("Set items"),
-                title_text =  _("Items per page"),
-                callback = function(spin)
-                    G_reader_settings:saveSetting("items_per_page", spin.value)
-                    self.ui:onRefresh()
+                keep_menu_open = true,
+                callback = function()
+                    local SpinWidget = require("ui/widget/spinwidget")
+                    local curr_items = G_reader_settings:readSetting("items_per_page") or 14
+                    local items = SpinWidget:new{
+                        width = Screen:getWidth() * 0.6,
+                        value = curr_items,
+                        value_min = 6,
+                        value_max = 24,
+                        ok_text = _("Set items"),
+                        title_text =  _("Items per page"),
+                        callback = function(spin)
+                            G_reader_settings:saveSetting("items_per_page", spin.value)
+                            --after changing items per page we reset the font size of item and set it as default
+                            G_reader_settings:saveSetting("items_font_size", math.floor(24 - ((spin.value - 6)/ 18) * 10 ))
```

Comment:
After changing items per page should I change font size for default or keep settings from `maximum font size for item`? Now I reset font size to default.

Reply:
It seems somewhat unexpected to change that when you're just changing one setting.

Reply:
Ok, I removed it.